syntax = "proto3";
package order;

service OrderService {
    rpc CreateOrder(CreateOrderRequest) returns (OrderResponse);
    rpc GetOrder(GetOrderRequest) returns (OrderResponse);
    rpc ReceiveMessage(MessageRequest) returns (MessageResponse);
}

message CreateOrderRequest {
    string user_id = 1;
    string product_id = 2;
    int32 quantity = 3;
}

message OrderResponse {
    string order_id = 1;
    string user_id = 2;
    string product_id = 3;
    int32 quantity = 4;
    string status = 5;
}

message GetOrderRequest {
    string order_id = 1;
}

message MessageRequest {
    string message = 1;
}

message MessageResponse {
    string status = 1;
}